src/index.ts

```diff
@@ -176,12 +180,23 @@ function forEmitOf<T = any>(emitter: SuperEmitter, options?: Options<T>) {
   const getRaceItems = raceFactory<T>(options, emitter);
 
   async function* generator() {
-    while (events.length || active) {
+    
+    function removeListeners() {
```

Comment:
Hey, could we define this function (`removeListeners`) up with `eventListener` & `endListener` ? it will help keep things consistent.